### PR TITLE
Fix/general improvements

### PR DESCRIPTION
**Pull request by MRigal**
Great code Kevin! I've made some general improvements which you may like. I will probably change more in a way that suits better to me on another branch

- Use consistently f-strings
- Enhance compartment level mapping
- Enhance usage of sets
- Avoid redundant code
- Avoid deep nesting with "with" statements (requirement Py 3.10+)
- Code linted and formatted using `ruff`
- Added minimal .gitignore
- Added support for more genders, more age categories and more reduction cards, using enums
- Added support to add/configure passengers via dataclass

### CODE REVIEW

**Comment by KevinW1998** [comment]
Hey there! Very cool, thanks a lot. I will check it out soon.

I know that my code might be a bit mashed together (also adjusted it to my needs), but any improvement towards a more clean API approach is welcomed!

**Comment by MRigal** [comment]
Still on it BTW, found another part, pushing soon!

**Comment by MRigal** [comment]
Done! :-)

**Comment by KevinW1998** [comment]
Thanks. I'll have a look at it in the next couple of days.

File: main.py [hunk 1]
```diff
@@ -126,150 +146,199 @@ def findOffersFiltered(self, station_from, station_to, day: datetime.date):
...
+    FEMALE = "female"
+    DIVERSE = "diverse"
+
+
+# Many more availables, but here probably the most important ones
+REDUCTION_CARDS = {
+    "DB-Bahncard-25-2Kl": 127,
+    "DB-Bahncard-50-2Kl": 129,
+    "DB-Ticket-Deutschland-2Kl": 9098153,
+    "Klimaticket": 100000042,
```
Comment (KevinW1998) [inline main.py]: Maybe it makes sense to have it as enum as well?

**Review by KevinW1998 — COMMENTED** [review]
Do you have any plans for more changes?

**Comment by KevinW1998** [comment]
Also, as far as I can see the required Python version is 3.10+, right? Maybe we should add it to the readme.

File: main.py (hunk 1 above)
Comment (MRigal) [reply main.py]: I could do that, indeed!

**Comment by MRigal** [comment]
Yes, I can add this in the README.

I've started to make bigger adaptations in another branch, but I've found a couple things more I could cherry-pick here over. I could add this to the two suggestions you've made and that would be it!

It's not a lot of time, but I'm not sure when I can do it, maybe tonight, maybe tomorrow...

**Comment by KevinW1998** [comment]
No hurry, just let me know when it is ready ^^

File: main.py (hunk 1 above)
Comment (MRigal) [reply main.py]: done

**Comment by MRigal** [comment]
Sooo, @KevinW1998 I think that's it, I don't have so much more to add from now on, I've also added the definition of passengers via dataclasses, with dict formatting as a function to make it cleaner. Let me know what you think about it! :-)

**Comment by MRigal** [comment]
One last possible enhancement would be to use Enums also for your AVAIL_LEVEL variables, but I won't do it tonight

**Comment by MRigal** [comment]
OK, @KevinW1998 now I really don't think that I have much more to enhance :-)

**Pull request merged**